packages/stryker-typescript/src/mutator/StringLiteralMutator.ts

```diff
@@ -10,39 +10,57 @@ export type AllStringLiterals =
   | ts.Token<ts.SyntaxKind.FirstTemplateToken>;
 
 export default class StringLiteralMutator extends NodeMutator<AllStringLiterals> {
-  name = 'StringLiteral';
+  public name = 'StringLiteral';
 
-  guard(node: ts.Node): node is AllStringLiterals {
-    return node.kind === ts.SyntaxKind.StringLiteral
-      || node.kind === ts.SyntaxKind.TemplateExpression
-      || node.kind === ts.SyntaxKind.FirstTemplateToken;
+  public guard(node: ts.Node): node is AllStringLiterals {
+    switch (node.kind) {
+      case ts.SyntaxKind.StringLiteral:
+      case ts.SyntaxKind.TemplateExpression:
+      case ts.SyntaxKind.FirstTemplateToken:
+        return true;
+      default:
+        return false;
+    }
   }
 
   private isInvalidParent(parent: ts.Node): boolean {
-    return parent.kind === ts.SyntaxKind.ImportDeclaration ||
-      parent.kind === ts.SyntaxKind.LastTypeNode ||
-      parent.kind === ts.SyntaxKind.JsxAttribute ||
-      parent.kind === ts.SyntaxKind.ExpressionStatement;
+    switch (parent.kind) {
+      case ts.SyntaxKind.ImportDeclaration:
+      case ts.SyntaxKind.LastTypeNode:
+      case ts.SyntaxKind.JsxAttribute:
+      case ts.SyntaxKind.ExpressionStatement:
+      case ts.SyntaxKind.LiteralType:
+        return true;
+      default:
+        return false;
+    }
   }
 
-  protected identifyReplacements(str: AllStringLiterals, sourceFile: ts.SourceFile): NodeReplacement[] {
+  protected identifyReplacements(str: AllStringLiterals): NodeReplacement[] {
     if (str.parent && this.isInvalidParent(str.parent)) {
       return [];
     }
 
-    if (
-      // Check for empty strings first.
-      (str.kind === ts.SyntaxKind.StringLiteral && str.text === '')
-      // Only check for the Token form of template literals. It's impossible to have a TemplateExpression
-      // that is empty as it's only used when there is a value embedded. I haven't found a case where the
-      // cast fails but the worst that can happen is it is undefined and we fall through to the wrong statement.
-      || (str.kind === ts.SyntaxKind.FirstTemplateToken && (str as ts.Node & { text: string }).text === '')
-    ) {
+    if (this.isEmptyString(str)) {
       return [{ node: str, replacement: '"Stryker was here!"' }];
-    }
-    else {
+    } else {
       return [{ node: str, replacement: '""' }];
     }
   }
 
+  private isEmptyString(str: AllStringLiterals) {
```

Comment:
I personally don't like early returns. Can't we do something like this?

```ts
return isEmptyString() || isEmptyTemplate();

function isEmptyString() {
   // Check if empty string
}

function isEmptyTemplate() {
   // check for is empty template
}
```

Reply:
Hmm, functions inside functions doesn't make it any better in my opinion, personally I don't really think early returns are confusing in this situation. If they ever become confusing we should extract some logic to seperate functions.

I would opt for something like:
```
private isEmptyString(str: AllStringLiterals) {
  let emptyString = true;

  function isEmptyString() {
     emptyString = false;
  }

  function isEmptyTemplate() {
    emptyString = false;
  }

  return emptyString;
}
```

Reply:
> functions inside functions doesn't make it any better in my opinion

Why not? Functional programming is popular and pretty common these days. It's both expressive and readable. Both JavaScript and TypeScript embrace these practices. It also adheres to the _clean code_. You could argue that the `isEmptyString` function does 3 things. Splitting it up will help with that. 

If all of that did not convince you you can still make 2 private methods here. One per check. It has largely the same benefits (although i would keep the functions private to the `isEmpty` method in this case).

Reply:
Well personally I don't really see the extra benefit because the case isn't all that complicated to begin with. But I will implement the suggested changes because it would be more future proof. I don't have a lot of experience with these kinds of constructs and their relation to clean code and they look nasty to me, but it seems like you've given sufficient thought to this so I'll just go with it :)

Reply:
In this case I would also prefer to have the local functions